### PR TITLE
s390x vectorization: implement atanh for complex vectorized data

### PR DESCRIPTION
**Pull request by AlekseiNikiforovIBM**
s390x vectorization: implement atanh for complex vectorized data

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/cpu/vec/vec256/zarch/vec256_zarch.h [hunk 1]
```diff
@@ -2319,7 +2323,7 @@ struct Vectorized<T, std::enable_if_t<is_zarch_implemented_complex<T>()>> {
     auto ln = Vectorized<T>(Vectorized<T>(b_a) + root).log();
     return Vectorized<T>(ln.vec().swapped()).conj();
 #else
-    return map(std::asin);
+    return mapOrdinary(std::asin);
```
Comment (jgong5) [inline aten/src/ATen/cpu/vec/vec256/zarch/vec256_zarch.h:2326]: This `#else` path seems never hit. Why?

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**